### DISCUSSION TITLE
Perpetual Mode

### DISCUSSION THREAD

**Discussion by well-it-wasnt-me** [zacharytomlinson/saspy#22] [Ideas]
Right now the lib will try to connect to a serial port until it succeed without ever stopping. 

Somehow this "bad" behaviour is ok in some real world use case. What about parametrize this thing ?

so around [here](https://github.com/zacharytomlinson/saspy/blob/27f7c54ee6f80796aa67cf7afc0bd2f285c5d09a/sas.py#L56) we put "if not perpetual" we put a break if fail.

**Comment by dedalgr** [comment]
sas.start() make this for you ???

На чт, 1.02.2024 г., 14:13 ч. Antonio ***@***.***> написа:

**Comment by dedalgr** [comment]
open whil check if you remove port and whil initial it again. You no need
while loop in __init__ it is a bad choice

На чт, 1.02.2024 г., 16:50 ч. Григор ***@***.***> написа:

**Reply by well-it-wasnt-me** [reply]
Yes. Sas.start (as it is now) will try to connect ad infinitum until a proper connection is made. 

While this is (somehow) good on real world use cases it might neeed be parametrized in order to not have a loop running forever without proper control

**Comment by dedalgr** [comment]
start method have reason. EMG load for 5 minute after power down, SMIB load
for 40 secounds. And SMIB initial serial, byt waith for EMG

На чт, 1.02.2024 г., 16:53 ч. Григор ***@***.***> написа:

**Comment by dedalgr** [comment]
Try to connect SMIB power to EMG, run as service and reboot EMG

На чт, 1.02.2024 г., 16:58 ч. Григор ***@***.***> написа:

**Comment by well-it-wasnt-me** [comment]
Ok. Then instead of a var parameter to initiate perpetual mode on/off we use a timer then. After 120 sec with cut it off. What you think ?
